src/components/hooks/__tests__/use-dynamic-presets.test.ts

```
@@ -0,0 +1,243 @@
+
+    const { result } = renderHook(() => useDynamicPresets())
+
+    await waitFor(() => {
+      expect(result.current).toEqual([])
+    })
+
+    expect(mockDb.getScheduledPosts).toHaveBeenCalled()
+  })
+
+  it('should return empty presets when db returns null', async () => {
+    ;(db as any).mockReturnValue(null)
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)